tests/basic/storage_pool_test.go

```diff
@@ -9904,3 +9904,255 @@ var _ = Describe("{AddDriveWithKernelPanic}", func() {
 	})
 
 })
+
+func selectPoolDeletableNode () (*node.Node){
+	var testNode *node.Node
+	
+	log.Info("Select non-kvdb node or node with >1 pools)")
+
+	stNodes := node.GetStorageNodes()
+
+	kvdbNodesIDs :=  []string{}
+	kvdbMembers, err := Inst().V.GetKvdbMembers(stNodes[0])
+	log.FailOnError(err, "Error getting KVDB members")
+
+	for _, n := range kvdbMembers {
+		kvdbNodesIDs = append(kvdbNodesIDs, n.Name)
+	}
+
+	for _, n := range stNodes {
+		if !Contains(kvdbNodesIDs, n.Id) {
+			testNode = &n 
+			break
+		} 
+	}
+	if testNode == nil {
+		testNode = &stNodes[0]
+		log.InfoD("cannot find nonkvdb node, select kvdb node %v for test", testNode.Addresses)
+		poolsMap, err := Inst().V.GetPoolDrives(testNode)
+		log.FailOnError(err, "cannot get pool drives")
+		log.InfoD("node %v has pools %+v", testNode.Addresses, poolsMap)
+		if len(poolsMap) <= 1 {
+			log.InfoD("try create new pool for test")
+			err = AddCloudDrive(*testNode, -1) 
+			log.FailOnError(err, "drive add failed")
+		}
+	} else {
+		log.InfoD("find non-kvdb storage node %v", testNode.Addresses)
+	}
+	dash.VerifyFatal(testNode != nil, true, "select test node")
+	return testNode
+}
+
+var _ = Describe("{PoolDeleteVariations}", func() {
+	 
+	/*
+		migrated from px-test: PoolDeleteVariations
+ 		1. Verify pool deletion after creating 50 volumes
+		2. Verify pool deletion after creating 30 snaps
+ 	*/
+	 
+	JustBeforeEach(func() {
+		StartTorpedoTest("PoolDeleteVariations", "Pool delete with variations", nil, 0)
+	})
+
+	var testNode *node.Node
+	BeforeEach(func() {
+		testNode = selectPoolDeletableNode()
+		dash.VerifyFatal(testNode != nil, true, "very if select test node ok")
+	  })
+	var contexts []*scheduler.Context
+	
+ 	numVolCreate := 50
+	itLog := fmt.Sprintf("1. Verify pool deletion after creating %v volumes", numVolCreate)
+	It(itLog, func() {
+		log.InfoD(itLog)
+		poolsMap, err := Inst().V.GetPoolDrives(testNode)
```

Comment:
`testNode.StoragePools[0]` can also work

Reply:
mmm, it will use a cached value, since I create and delete the pool, the StoragePools will be obsoleted.

Reply:
But then in the `PoolDeleteServiceDisruption` test you are using `StoragePools` to get the pool to delete. Maybe change it there? Or we can call `Inst().S.RefreshNodeRegistry()` in `JustBeforeEach` for the tests you and I add.

Reply:
I would prefer the latter approach so that the node registry is always updated after each test.